packages/chakra-ui-core/src/Editable/tests/Editable.test.js

```diff
@@ -0,0 +1,153 @@
+import { Editable, EditableInput, EditablePreview } from '../../'
+import { render, defaultProviders, userEvent, fireEvent } from '@/tests/test-utils'
+import { useId } from '../../utils'
+
+// mocks
+jest.mock('@/packages/chakra-ui-core/src/Toast/index.js', () => {})
+jest.mock('@/packages/chakra-ui-core/src/utils/generators.js')
+
+const renderComponent = (props) => {
+  const inlineAttrs = (props && props.inlineAttrs) || ''
+  const base = {
+    components: { Editable, EditableInput, EditablePreview },
+    provide: () => defaultProviders(),
+    template: `<Editable ${inlineAttrs}>
+      <EditablePreview data-testid="preview" />
+      <EditableInput data-testid="input" />
+    </Editable>
+    `,
+    ...props
+  }
+  return render(base)
+}
+
+it('should render correctly', async () => {
+  const inlineAttrs = 'defaultValue="testing" '
+  const { asFragment } = renderComponent({ inlineAttrs })
+
+  expect(asFragment(document.body.innerHTML)).toMatchSnapshot()
+})
+
+it('should render correctly - input', async () => {
+  useId.mockReturnValueOnce('1')
+  const inlineAttrs = 'defaultValue="testing" '
+  const { asFragment, getByTestId } = renderComponent({ inlineAttrs })
+
+  const preview = getByTestId('preview')
+  await userEvent.click(preview)
+
+  expect(asFragment(document.body.innerHTML)).toMatchSnapshot()
+})
+
+it('uncontrolled: handles callbacks correctly', async () => {
+  const onChange = jest.fn()
+  const onCancel = jest.fn()
+  const onSubmit = jest.fn()
+  const onEdit = jest.fn()
+
+  const inlineAttrs = '@edit="onEdit" @change="onChange" @cancel="onCancel" @submit="onSubmit" defaultValue="Hello "'
+  const { getByTestId } = renderComponent({ inlineAttrs, methods: { onChange, onCancel, onSubmit, onEdit } })
+
+  const preview = getByTestId('preview')
+  await userEvent.click(preview)
+
+  const input = getByTestId('input')
+
+  // calls `onEdit` when preview is focused
+  fireEvent.focus(preview)
+  expect(onEdit).toHaveBeenCalled()
+
+  // calls `onChange` with input on change
+  userEvent.type(input, 'World')
+  expect(onChange).toHaveBeenCalledWith('Hello World')
+
+  // calls `onCancel` with previous value when "esc" pressed
+  fireEvent.keyDown(input, { key: 'Escape' })
+  expect(onCancel).toHaveBeenCalledWith('Hello ')
+
+  // calls `onSubmit` with previous value when "enter" pressed after cancelling
+  fireEvent.keyDown(input, { key: 'Enter' })
+  expect(onSubmit).toHaveBeenCalledWith('Hello ')
+
+  // TODO: ⚠️ we should make controlledInput for vue
+  // remove this line, previous input value stays and next 2 expect fails
+  input.value = 'Hello '
```

Comment:
Doesn't this directly manipulate the DOM input and not the value props for controlled component? What did you have in mind for this?

Reply:
Did you remove this line and run the test? you will understand what i mean.

Reply:
Right now I can't checkout to your fork, but I think it's something we can look into in the future since it's in the testing paradigm. The component works as expected. Let me know if you have any thoughts about this.

Reply:
Yep component works. It was about controlled and uncontrolled inputs. `event.target.value` shouldnt change if its controlled etc. Its not an issue for now. We can take a look in the future.